src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.Process.cs

```diff
@@ -32,35 +35,73 @@ private static RemoteInvokeHandle RemoteInvoke(MethodInfo method, string[] args,
-                // to be profiled / have code coverage, so we remove these environment variables for that process 
-                // before it's started.
-                psi.Environment.Remove("Cor_Profiler");
-                psi.Environment.Remove("Cor_Enable_Profiling");
-                psi.Environment.Remove("CoreClr_Profiler");
-                psi.Environment.Remove("CoreClr_Enable_Profiling");
-            }
+                // Start the other process and return a wrapper for it to handle its lifetime and exit checking.
+                var psi = options.StartInfo;
+                psi.UseShellExecute = false;
 
-            // If we need the host (if it exists), use it, otherwise target the console app directly.
-            string testConsoleAppArgs = "\"" + a.FullName + "\" " + t.FullName + " " + method.Name + " " + string.Join(" ", args);
+                if (!options.EnableProfiling)
+                {
+                    // Profilers / code coverage tools doing coverage of the test process set environment
+                    // variables to tell the targeted process what profiler to load.  We don't want the child process 
+                    // to be profiled / have code coverage, so we remove these environment variables for that process 
+                    // before it's started.
+                    psi.Environment.Remove("Cor_Profiler");
+                    psi.Environment.Remove("Cor_Enable_Profiling");
+                    psi.Environment.Remove("CoreClr_Profiler");
+                    psi.Environment.Remove("CoreClr_Enable_Profiling");
+                }
 
-            if (!File.Exists(TestConsoleApp))
-                throw new IOException("RemoteExecutorConsoleApp test app isn't present in the test runtime directory.");
+                // If we need the host (if it exists), use it, otherwise target the console app directly.
+                string testConsoleAppArgs = "\"" + a.FullName + "\" " + t.FullName + " " + method.Name + " " + string.Join(" ", args);
 
-            psi.FileName = HostRunner;
-            psi.Arguments = ExtraParameter + testConsoleAppArgs;
+                if (!File.Exists(TestConsoleApp))
+                    throw new IOException("RemoteExecutorConsoleApp test app isn't present in the test runtime directory.");
 
-            // Return the handle to the process, which may or not be started
-            return new RemoteInvokeHandle(options.Start ?
-                Process.Start(psi) :
-                new Process() { StartInfo = psi }, options);
+                psi.FileName = HostRunner;
+                psi.Arguments = ExtraParameter + testConsoleAppArgs;
+
+                    // Return the handle to the process, which may or not be started
+                    return new RemoteInvokeHandle(options.Start ?
+                        Process.Start(psi) :
+                        new Process() { StartInfo = psi }, options);
+            }
+            else
+            {
```

Comment:
> Do you know what the problem is with Debugger.Launch?

https://github.com/dotnet/coreclr/issues/12074

> I'm happy to do anything so long as it makes a better experience

Understood, I'm just saying while I'm get that this might improve the experience when you're looking at an individual test and know it's safe to run in the current process in isolation, a) even for individual tests for which it's "safe" it's very likely not when running multiple tests (after all, there's a reason it's being remotely invoked), and b) there are lots of tests for which it won't work, and I fear that will actually make the overall experience worse.

Up to you, though.